sshnp_docker_demo/README.md

```diff
@@ -0,0 +1,262 @@
+# sshnp_docker_demo
+
+This is a demo of sshing into one docker container from another docker container without having ports open (not even port 22). There are three docker containers: `sshnp`, `sshnpd`, and `sshrvd`. This demo involves using [sshnp](https://github.com/atsign-foundation/sshnoports).
+
+## Getting Started
+
+Follow these steps to get sshnp working on your machine using Docker.
+
+### 1. Prerequisites
+
+1. Install [Docker](https://www.docker.com/products/docker-desktop/) and open it in the background. Check that you have docker installed. Run this in your terminal. Your output should be similar:
+
+```sh
+docker --version
+Docker version 23.0.5, build bc4487a
+```
+
+2. You will need 3 atSigns and their associated `.atKeys` files (one for each docker container). For each of your atSigns, put the `.atKeys` file into the `keys/` directory. For example, your file structure should be similar to:`sshnp/keys/@sshnp_key.atKeys`, `sshnpd/keys/@sshnpd_key.atKeys`, `sshrvd/keys/@sshrvd_key.atKeys`.
+
+### 2. Finding IP of sshrvd
+
+This is a tedious but short step where we need to do to find the IP of the sshrvd docker container.
+
+Let's build and run the `sshrvd` docker container by running
+
+```sh
+cd sshrvd
+sh db.sh
+```
+
+Now, in another terminal window, find the IP address of the `sshrvd` docker container.
+
+```sh
+docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' sshrvd
+172.**.*.*
+```
+
+Note this IP address for later.
+
+Stop the docker container with `Ctrl + D`
+
+```
+root@7452c3c9e744:/atsign# 
+exit
+```
+
+### 3. Setting up the Shell Scripts
+
+1. Edit shell scripts `.startup.sh` in each of the three folders. (A) - sshnp, (B) - sshnpd, (C) - sshrvd.
+
+(A) The `sshnp/.startup.sh` script:
+
+- replace `john@example.com` with your email
```

Comment:
Can we use an interactive script to do TOKEN substitution?

Reply:
This is what I came up with:

![image](https://github.com/atsign-foundation/at_demos/assets/79019866/dc81ac33-4eac-4e6e-8b39-9f0b44490cef)
![image](https://github.com/atsign-foundation/at_demos/assets/79019866/84b522aa-83ed-444a-817c-e8811ab31c6f)

Generates this:

![image](https://github.com/atsign-foundation/at_demos/assets/79019866/fbbaff9f-21ff-4de8-bcb4-e00edaf2966c)

Reply:
I had some trouble naming the script, though - 
![image](https://github.com/atsign-foundation/at_demos/assets/79019866/d9106056-f31b-4794-8a81-6cfd76ac924a)